gsutil.py

```diff
@@ -73,24 +73,39 @@ def OutputAndExit(message):
     ('pyasn1-modules', ''),  # oauth2client dependency
     ('rsa', ''),  # oauth2client dependency
     ('apitools', ''),
-    ('boto', ''),
     ('gcs-oauth2-boto-plugin', ''),
     ('fasteners', ''), # oauth2client and apitools dependency
     ('monotonic', ''), # fasteners dependency
     ('httplib2', 'python2'),
     ('python-gflags', ''),
     ('retry-decorator', ''),
-    ('six', ''),
+    ('six', ''), # Python 2 / 3 compatibility dependency
     ('socksipy-branch', ''),
 ]
-for libdir, subdir in THIRD_PARTY_LIBS:
-  if not os.path.isdir(os.path.join(THIRD_PARTY_DIR, libdir)):
-    OutputAndExit(
-        'There is no %s library under the gsutil third-party directory (%s).\n'
-        'The gsutil command cannot work properly when installed this way.\n'
-        'Please re-install gsutil per the installation instructions.' % (
-            libdir, THIRD_PARTY_DIR))
-  sys.path.insert(0, os.path.join(THIRD_PARTY_DIR, libdir, subdir))
+
+# These are special third party libraries that live in
+# `gsutil/third_party/vendored`. These libraries have modifications specifically
+# for gsutil to operate smoothly that may not yet be present in the latest
+# release of the package.
+VENDORED_THIRD_PARTY_LIBS = [
+    ('boto', '') # XML API and AWS interop dependency
+]
+
+
+def PrependThirdPartyLibToPath(path_component_groups, component_path):
```

Comment:
Semantic nit: "component_path" doesn't make much sense to me - as a path is composed of its components.  Maybe "parent_path"?

Reply:
SGTM, updated :+1: